pymc/distributions/discrete.py

```diff
@@ -813,6 +813,12 @@ def dist(cls, p, *args, **kwargs):
         p = at.as_tensor_variable(floatX(p))
         return super().dist([p], *args, **kwargs)
 
+    def get_moment(rv, size, p):
```

Comment:
We were using the mode before (always return 1 regardless of `p`) 

Not sure if we should change the default to be the rounded mean as in this PR. Do you have a good reason for it?

Reply:
It was mentioned in #5078 that "the "higher-order" one is the most useful (that is mean > median > mode)" that's why I went with mean. Apart from that I don't really have any good reason. Let me know if I should change it back to mode.

Reply:
Yeah in general the mean is better but not necessarily so for discrete distributions. The geometric has the additional issue that it can blow up quite often. 

Maybe someone else will be able to ping in here...

Reply:
If nobody does, my guy feeling would be to take the conservative choice and keep things as they were in V3 (that is using the mode)

Reply:
Changed my mind, let's use this and see if anybody complains :D. You still need to solve the merge conflicts that cropped in due to other merged PRs